### PR TITLE
add feature to view fixture source location in invocations with --fixtures-per-test option

### PR DESCRIPTION
**Pull request by rahul-kumi**
This implements the proposal in issue #8606

The following are the changes made to the behaviour:
- Always showing the path to the fixtures when using the `--fixtures-per-test` option
- Only showing the first line of the docstrings unless `--verbose` is given

### CODE REVIEW

**Comment by rahul-kumi** [comment]
I have not made an entry into `changelog` and `documentation`.
I have put them on hold to get your reviews on these change and add them.

@The-Compiler, @nicoddemus Please review the PR and let me know if there are changes to be made.

File: src/_pytest/python.py
Comment (The-Compiler) [inline src/_pytest/python.py:1433]: IMHO this gets just a tad more readable if you [eschew the extraneous else](https://www.youtube.com/watch?t=289&v=JVVMMULwR4s&feature=youtu.be) :wink:

```suggestion
        bestrel = get_best_relpath(fixture_def.func)
        funcargspec = f"{argname} -- {bestrel}"
```

File: testing/python/show_fixtures_per_test.py
Comment (The-Compiler) [inline testing/python/show_fixtures_per_test.py:175]: seems like an unintended/unrelated change?

**Review by The-Compiler — CHANGES_REQUESTED** [review]
Yay, thank you! :tada: Looks good overall, I commented on two small things.

**Comment by rahul-kumi** [comment]
@The-Compiler, Thanks for the review.
I made the suggested changes.

Can you please take a look and let me know if documentation and changelog for this change can be added?

**Review by The-Compiler — COMMENTED** [review]
Thanks! One thing I noticed just now is that it kind of breaks with docstrings where the summary spans across multiple lines, like e.g. with `mocked_urlopen`:

https://github.com/pytest-dev/pytest/blob/3a2fd9630547e224bea6e024a4f6852761235c61/testing/test_pastebin.py#L128-L132

This now displays as:

```
-------------------- fixtures used by test_create_new_paste --------------------
------------------------ (testing/test_pastebin.py:163) ------------------------
mocked_urlopen -- testing/test_pastebin.py:129
    Monkeypatch the actual urlopen calls done by the internal plugin
monkeypatch -- src/_pytest/monkeypatch.py:29
    A convenient fixture for monkey-patching.
pastebin -- testing/test_pastebin.py:88
    no docstring available
```

so ideally, we'd instead split on the first empty line. Perhaps something like `.split('\n\n')` would already be good enough?

Also, it looks like this PR only adjusts `--fixtures-per-test` - I think it'd be good to adjust `--fixtures` in the same way as well.

**Comment by rahul-kumi** [comment]
@The-Compiler,
Use of the `.split('\n\n')` seems to be a good idea to tackle cases where the docstring sentence is spread across different lines. Introducing this would mean that we would be returning the first paragraph of the docstring if it is paragraphed else the entire content.

Hope this behaviour is acceptable.

**Comment by rahul-kumi** [comment]
@The-Compiler,
>so ideally, we'd instead split on the first empty line. Perhaps something like .split('\n\n') would already be good enough?

This is addressed in this commit: c0dc982

>I think it'd be good to adjust --fixtures in the same way as well.

This is addressed in this commit: 02f026f

**Review by The-Compiler — CHANGES_REQUESTED** [review]
Thanks for the update! Also a great idea to add some test for the docstring parsing.

One thing I didn't realize before looking at the updated diff is that `--fixtures --verbose` actually did show the path in a different color:

image

I think this helps readability a bit, and IMHO it'd be good to keep that. What do you think?

Other than that, this will still need a changelog entry, but seems fine otherwise. I don't think any other changes to the documentation are needed, but maybe I missed something.

**Comment by rahul-kumi** [comment]
@The-Compiler, I have made the suggested changes.
Can you please review the same and let me know if there's anything else that has to be improved.

**Comment by The-Compiler** [comment]
I've seen your review request and will take another look, but it'll probably be somewhen next week until I get around to it.

File: src/_pytest/python.py
Comment (The-Compiler) [inline src/_pytest/python.py:1511]: Seems like an unrelated and potentially unneeded change? (but I'm okay with both variants).

File: src/_pytest/python.py
Comment (The-Compiler) [inline src/_pytest/python.py:1513]: Is this intended?

**Review by The-Compiler — CHANGES_REQUESTED** [review]
Thanks, the output looks great now! I commented on two changes where I'm unsure what your intentions were - after that I think this should be ready to merge!

File: src/_pytest/python.py
Comment (rahul-kumi) [reply src/_pytest/python.py:1511]: I wanted to keep this consistent, noticed a similar check taking the `startswith` form. So made this change.

File: src/_pytest/python.py
Comment (rahul-kumi) [reply src/_pytest/python.py:1513]: Yes this inclusion is intended.

Comment (The-Compiler) [reply src/_pytest/python.py:1513]: Alright - I don't really see the point of using an f-string here (with only one single string variable in it), but this is too minor to block this PR.

**Review by The-Compiler — APPROVED** [review]
Thanks again! Makes this output much more useful, in my opinion!

File: src/_pytest/python.py
Comment (rahul-kumi) [inline src/_pytest/python.py:1432]: @The-Compiler, I noticed your suggestion to use `argname` in this line as a direct variable reference instead of using it as a formatted string. I am not able to find that comment to reference it.

I think that would be a good thing to do and keeps things consistent. Please let me know if this small change can go in as a separate PR.

**Pull request merged**
